src/extension.c

```diff
@@ -149,7 +135,13 @@ extension_is_loaded(void)
 	 */
 	id = get_extension_oid(EXTENSION_NAME, true);
 
-	if (OidIsValid(id))
+	/*
+	 * the proxy table check is necessary because during an extension drop the
+	 * proxy table will be deleted before the extension itself. You don't want
+	 * to mark the extension as CREATED in this case
+	 */
+
+	if (OidIsValid(id) && proxy_table_exists())
```

Comment:
Btw, I think the reason for returning `true` (or some intermediate state) was that the cache invalidation events need to work when dropping, since there is an `extension_is_loaded()` check in the cache invalidation function. Now, this might still happen to work (again incidentally) because the other proxy tables are dropped before the extension proxy table, or something like that. If we are moving to a boolean state machine and always returning false during extension drops (again you are only returning false after the proxy table is dropped which happens in a largely unknown point after drop starts) then we also need to make sure that cache invalidation works, and not only happens to work because tables happen to be removed in some order.